src/server/pmix_server.c

```diff
@@ -882,6 +882,11 @@ static void _deregister_client(int sd, short args, void *cbdata)
             if (NULL != (peer = (pmix_peer_t*)pmix_pointer_array_get_item(&pmix_server_globals.clients, info->peerid))) {
                 pmix_psensor.stop(peer, NULL);
             }
+            /* ensure we close the socket to this peer so we don't
+             * generate "connection lost" events should it be
+             * subsequently "killed" by the host */
+            CLOSE_THE_SOCKET(peer->sd);
```

Comment:
Note this part of the patch is wrong. The `CLOSE_THE_SOCKET` needs to be in the selection statement for the NULL protection.